### DISCUSSION TITLE
Missing declarations in my application

### DISCUSSION THREAD

**Discussion by yonagy82** [khoih-prog/ESP32TimerInterrupt#24] [General]
Hello,
I'm testing the library using the example ISR_16_Timars_Array. While compiling I get errors of missing declarations about the method _**timer_isr_callback_add**_ and the parameters **_TIMER_INTR_T0_** and **_TIMER_INTR_T1_**. I assume I miss some file inclusion - which one?

And another question - I'm using the board Heltec WiFILoRa 32(V2). To minimize interference with other HW/SW fwatures I prefer to use timer 2 - is it possible?

Best regards
Yons Gy

**Comment by khoih-prog** [comment]
1. **TIMER_INTR_T0 and TIMER_INTR_T1**
It's better to do some little research around, especially the previous issues, such as [**TIMER_INTR_0 Not defined** #23](https://github.com/khoih-prog/ESP32TimerInterrupt/issues/23), and save time for both of us.

2. Using Timer2-Timer3

It's completely possible, assuming **Heltec WiFILoRa 32(V2)** has correct firmware and configurations.

For example, instead of using https://github.com/khoih-prog/ESP32TimerInterrupt/blob/668316d71417bcb9bb31abcdedd154fff77cea7b/examples/Change_Interval/Change_Interval.ino#L103-L104

use

```
// Init ESP32 timer 2
ESP32Timer ITimer2(2);
...
```

**Comment by yonagy82** [comment]
Thank you very much
